controller/k8s/watcher_test.go

```diff
@@ -51,4 +58,34 @@ func TestWatcher(t *testing.T) {
 			t.Fatalf("Expected error, got nil")
 		}
 	})
+	t.Run("Returns error if the watcher does not initialize in the time limit", func(t *testing.T) {
```

Comment:
This test name is identical to the preceding test, and it doesn't really reflect what's being tested in this test. I think you want something along the lines of:

> Retries establishing the watch if it fails to establish the first time